types.ts

```diff
@@ -970,6 +977,45 @@ export type VariousSupportedAuthentication =
  */
 export type VariousSupportedAuthenticationTypes = $Values<Pick<VariousSupportedAuthentication, 'type'>>;
 
+/**
+ * @hidden
+ */
+export enum ReservedAuthenticationKeys {
+  /**
+   * References the default user authentication of the pack.
+   */
+  Default = 'DEFAULT',
+
+  /**
+   * References the system authentication of the pack.
+   */
+  System = 'SYSTEM',
+}
+
+/**
+ * TODO(patrick): Unhide this.
+ * @hidden
+ */
+export interface AuxiliaryAuthentication {
```

Comment:
I can't quite articulate it but I feel like we should either name/frame this as being specific to admins, or add a field to indicate that this definition applies only to admins and force you to set it for now.

We will only offer these to org admins in Brain today. If for some reason we support auxiliary auth for non-admin users, things would break badly if the auth types offered for admins started to get offered to non-admins. So if we're not naming the whole concept in a way to force it to always relate to admins I think we at least need to add a restriction that forces this current crop of defs to be tied to admins if we think we may extend what this is for in the future.

Reply:
Calling them "Admin authentications" is growing on me. But I'll throw out one other brainstorm idea:

"organization authentications" - all our organization-level settings are only available admins anyways. I think this makes it pretty clear these will not be used in classic doc usage context. Potential downsides:
- Have we discussed whether we'll allow non-enterprise customers to use meta syncs without making organizations? This assumes the implementation decision of always using organizations, which might not be safe.
- If we are aiming towards supporting a shared slack bot writing connection, we might want that to work at the workspace level, not org level?

Reply:
Yeah it's the Slack but case that makes me rule out "organization authentication". I think "admin" captures that a bit better, because workspaces have admins too and so we maintain some useful ambiguity there. Still an open question if "admin" will always be an appropriate descriptor for these.

We may buy ourselves a little more extensibility if we leave these as "auxiliary auth" and add an `audience` property or something that for now is only allowed to be `audience: AuxiliaryAuthenticationAudience.OrgAdmin` or `CodaBrainAdmin` until we add further support.

Reply:
I hear you... but I'm worried that direction will cut off a different form of flexibility. Imagine we scrap Coda Brain and instead develop a different kind of shared workspace tables product feature. I feel like we'd want to keep an "audience" fully as vague as "admin" for that reason.